src/17-infinite-discovery/03-artworks-mutli2vec.ts

```diff
@@ -0,0 +1,183 @@
+import weaviate, { generateUuid5 } from "weaviate-ts-client"
+import _ from "lodash"
+import { GravityArtwork } from "./types"
+import { deleteIfExists } from "system/weaviate"
+import dotenv from "dotenv"
+import { getArtworks, getImageDataForArtworks } from "./helpers"
+
+dotenv.config()
+
+const CLASS_NAME = "InfiniteDiscoveryArtworks"
+const BATCH_SIZE = 2
+
+const client = weaviate.client({
+  host: process.env.WEAVIATE_URL!,
+})
+
+async function main() {
+  await deleteIfExists(CLASS_NAME)
+  await prepareArtworkCollection()
+  const artworks = await getArtworks()
+  await insertArtworks(artworks)
+}
+
+/**
+ * Create and configure a new collection to hold artworks
+ */
+async function prepareArtworkCollection() {
+  const classWithProps = {
+    class: CLASS_NAME,
+    vectorizer: "multi2vec-clip",
+    moduleConfig: {
+      "multi2vec-clip": {
+        textFields: [
+          "title",
+          "medium",
+          "materials",
+          "categories",
+          "tags",
+          "additionalInformation",
+        ],
+        imageFields: ["image"],
```

Comment:
Noting the config determining what to vectorize is slightly different when using `multi2vec-clip`.